# Solar heating model 1

### 1. Physical constants, run parameters and model parameters
We start by setting physical constants, run parameters that control the model simulation and model parameters that determine the characteristics of the model:

In [5]:
% Set physical constants
pureWaterFreezingPoint  = 273.15;  % pure water freezing point/K

% Set run parameters
deltaT     =    86400.0;           % time step/s
startTime  =    0.0;               % time to start integration/s
nTimeSteps =  365;                 % total number of time steps

% Set model parameters
solarConstant         = 1360.8;    % solar constant/(W m-2)
effectiveHeatCapacity = 2.9526E08; % effective heat capacity of the 
                                   % atmosphere-ocean system/(J m-2 K-1)

### 2. Initial conditions
Next we specify the initial temperature and initial time:

In [8]:
% Set initial conditions
temperature = 15.0 + pureWaterFreezingPoint;
myTime      = startTime;

### 3. Time loop
Finally, we add the time loop in terms of a `for` loop. It keeps track of the simulated time and approximates the time-dependent evolution of temperature using the Euler method:

In [9]:
% Time loop (myIter = iteration counter)
for myIter=1:nTimeSteps
    
    % Record simulated time
    myTime = myTime + deltaT;
    
    % Step global annual-mean surface temperature forward in time 
    % (by "Euler forward" or "forward-in-time" method)
    temperature  = temperature + deltaT/effectiveHeatCapacity* ...
                                 (solarConstant/4.0);

end % end of time loop

Note that we need only one temperature variable, which is assigned a new value every time the body of the `for` loop is executed. After the last time step, this variable contains the final temperature value:

In [10]:
temperature

temperature =  324.49


Hence the temperature after one year is 324.49 K. To find out how long it takes to reach the boiling point or how the temperature change would proceed if time went on, you could simply reduce or increase the number of time steps `nTimeSteps` and rerun your M-file script `solar_heating1.m`, or alternatively, rerun the cells in this notebook. 

## Summary
M-files can be used to carry out complex tasks in a concise manner, much simpler than in a spreadsheet.